### ISSUE TITLE
Autonym names not presented appropriately with authors

### ISSUE THREAD

**Issue by m-hope** [AtlasOfLivingAustralia/profile-hub#410]
Autonyms (subspecies names that have the same name as the species E.g. Adenanthos cygnorum cygnorum) are not being presented appropriately with authors in the profile name.

Eg: if you use the example above in the ALA search, it will present the name as: "Adenanthos cygnorum Diels subsp. cygnorum", but if you create a profile it will match the name to "Adenanthos cygnorum subsp. cygnorum", ie: the author ('Diels') is missing.

**Comment by RobinaSanderson** [comment]
Is the issue with the ALA name index, the renaming of the profile or the source name from APNI.

**Comment by m-hope** [comment]
I think the problem is in Profiles, specifically handling of autonym names by Profile, as evidenced by the following bug:

In a collection create a profile for Cerastium fontanum subsp. fontanum which will match to the same name, then create another profile for Cerastium fontanum Baumg. subsp. fontanum which also matches to the same name.  If you now search for either Cerastium fontanum Baumg. subsp. fontanum  or Cerastium fontanum Baumg. subsp. fontanum you will now get two search results (both listed as Cerastium fontanum subsp. fontanumbut they both point to a single profile (the last one that was created). This is because in nomenclatural terms these two profiles names are the same entity.  In fact the second profile should not have been created (try creating two profiles one called Moraceae Gaudich. and one called Moraceae Gaudich., the second one should be rejected because Profiles recognises it is the same as the first one).

**Comment by ben3000** [comment]
If I try to create a new profile and use "Adenanthos cygnorum subsp. cygnorum" as the Profile Name, I can see that https://github.com/AtlasOfLivingAustralia/profile-service/blob/dev/grails-app/services/au/org/ala/profile/NameService.groovy#L56 receives a matching name that has no authority property. The workaround code at https://github.com/AtlasOfLivingAustralia/profile-service/blob/dev/grails-app/services/au/org/ala/profile/NameService.groovy#L104 then tries to find the authority but can't (because I didn't provide one, and ALANameSearcher didn't either).

I also tried to create a new profile using "Adenanthos cygnorum Diels subsp. cygnorum", but the above-mentioned workaround code doesn't detect the authority.

My profiles instance is running the latest (20170413) namematching index. I thus think the problem lies deeper than Profiles. If the name searcher won't provide the relevant data, Profiles isn't going to be able to do anything useful with it.

**Comment by ben3000** [comment]
Fixed a bug in AtlasOfLivingAustralia/profile-service@e5e0cc0893f54a0005ff5f116e32b9945eb3f9eb that relates to this issue.

**Comment by m-hope** [comment]
@ben3000, you are correct... the problem is deeper than Profiles... I discovered a similar issue when I was looking into a couple of other names related issues... I think Profiles assumes that the authority is always at the end of the name string (and for 99.99% of the time it is correct), but there are a couple of cases (autonyms and this profile: https://profiles-dev.ala.org.au/opus/design-test/profile/Cynoglossum%20sp.%20Inland%20Ranges%20(C.A.%20Gardner%2014499) ) don't.   In looking at the JSON return from NSL, I don't think they handle it either (in the case of autonyms the scientificNameAuthorship field is blank)... so at the present time there is no easy fix for this problem.

I'll confirm this with the guys from ABRS today.

**Comment by ben3000** [comment]
AtlasOfLivingAustralia/ala-name-matching#1 seems like the most relevant currently-available bug in that project.

**Comment by ben3000** [comment]
@m-hope Yes, and also see how the _Cynoglossum_ name _should_ be formatted: https://florabase.dpaw.wa.gov.au/browse/profile/31757. The BIE has the same problem, italicising the phrase name portion of the name: http://bie.ala.org.au/species/http://id.biodiversity.org.au/node/apni/2896860.
